### ISSUE TITLE
Circular Reference Detected when serializing log record

### ISSUE THREAD

**Issue by swenzel** [Delgan/loguru#286]
When running the following python script:

```python
import sys
from loguru import logger

config = {
    "handlers": [
        {"sink": sys.stdout, "format": "{time} - {message}", "serialize": True, "colorize": True},
    ],
}
logger.configure(**config)
with logger.catch():
    raise RuntimeError("Something happened")
```
I get:
```plaintext
--- Logging error in Loguru Handler #1 ---
Record was: {'elapsed': datetime.timedelta(microseconds=21630), 'exception': (type=<class 'RuntimeError'>, value=RuntimeError('Something happened'), traceback=<traceback object at 0x7f981875cbe0>), 'extra': {'record': {...}}, 'file': (name='test.py', path='test.py'), 'function': '<module>', 'level': (name='ERROR', no=40, icon='❌'), 'line': 11, 'message': "An error has been caught in function '<module>', process 'MainProcess' (80051), thread 'MainThread' (140291248607616):", 'module': 'test', 'name': '__main__', 'process': (id=80051, name='MainProcess'), 'thread': (id=140291248607616, name='MainThread'), 'time': datetime(2020, 6, 12, 12, 12, 51, 845339, tzinfo=datetime.timezone(datetime.timedelta(seconds=7200), 'CEST'))}
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    raise RuntimeError("Something happened")
RuntimeError: Something happened

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/.local/share/virtualenvs/test-TSP576Db/lib/python3.7/site-packages/loguru/_handler.py", line 165, in emit
    formatted = self._serialize_record(formatted, record)
  File "/home/user/.local/share/virtualenvs/test-TSP576Db/lib/python3.7/site-packages/loguru/_handler.py", line 253, in _serialize_record
    return json.dumps(serializable, default=str) + "\n"
  File "/home/user/.asdf/installs/python/3.7.7/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/user/.asdf/installs/python/3.7.7/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/home/user/.asdf/installs/python/3.7.7/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
ValueError: Circular reference detected
--- End of logging error ---
```
I'm using `loguru==0.5.0` and `python==3.7.7`.
Looks like the problem is that `record["extra"]` has an item `"record"` which points back to `record` so that `record is record["extra"]["record"]` which is used in the object that shall be serialized and causes a circular reference within `_serialize_record` here:
https://github.com/Delgan/loguru/blob/8c425111390f7c2e8cec3185d18ac9ab111dc0b0/loguru/_handler.py#L235

**Comment by Delgan** [comment]
Thanks for the report!

It's a new bug in `0.5.0`:  while logging a message like `logger.info("Log message", foobar=123)` the ``**kwargs`` are now automatically added to the `extra` dict. However, when an error is captured and logged by `catch()`, the `record` is added to the ``**kwargs`` so that useful information can be displayed (e.g. `"An unexpected error occurred in thread {record[thread]}"`). The same problem would occur using `logger.opt(record=True).info("It's a bug")`.

I'll fix it and publish a new version later today, thank you!

**Comment by swenzel** [comment]
> Thanks for the report!

You're welcome, this is the least I can do :slightly_smiling_face:
Thanks for this awesome logging library, keep up the good work!

**Comment by Delgan** [comment]
>  Thanks for this awesome logging library, keep up the good work!

Thank you! :smiley: 

I published `v0.5.1`, the error should be gone now. :+1:
